### PR TITLE
Modernise tests in `test_create.py` 

### PR DESCRIPTION
**Pull request by ForgottenProgramme**
### Description

Draft PR for easy review.

Resolves some work listed in this epic: 
- https://github.com/conda/conda/issues/13292

### Checklist - did you ...

- [ ] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: tests/test_create.py [hunk 1]
```diff
@@ -2177,94 +2177,105 @@ def side_effect(self, url, **kwargs):
...
-            assert package_is_installed(prefix, "pycosat")
-            assert package_is_installed(prefix, "conda-build")
+def test_dont_remove_conda_1(
+    monkeypatch: MonkeyPatch, tmp_env: TmpEnvFixture, conda_cli: CondaCLIFixture
+):
+    with tmp_env() as prefix:
+        monkeypatch.setenv("CONDA_ROOT_PREFIX", prefix)
+        reset_context()
+        assert context.root_prefix == str(prefix)
+        _, _, _ = conda_cli(
```
Comment (kenodegard) [inline tests/test_create.py:2200]: I like to leave a little extra space and no need to store the output if we don't use it:

```suggestion

        conda_cli(
```

File: tests/test_create.py [hunk 2]
```diff
@@ -2177,94 +2177,105 @@ def side_effect(self, url, **kwargs):
...
-            assert package_is_installed(prefix, "conda")
-            assert package_is_installed(prefix, "pycosat")
+    with tmp_env() as prefix:
+        monkeypatch.setenv("CONDA_ROOT_PREFIX", prefix)
+        reset_context()
+        assert context.root_prefix == str(prefix)
 
-            with pytest.raises(CondaMultiError) as exc:
-                run_command(Commands.REMOVE, prefix, "pycosat")
+        _, _, _ = conda_cli("install", f"--prefix={prefix}", "conda", "--yes")
```
Comment (kenodegard) [inline tests/test_create.py:2232]:
```suggestion
        conda_cli("install", f"--prefix={prefix}", "conda", "--yes")
```

File: tests/test_create.py [hunk 3]
```diff
@@ -2177,94 +2177,105 @@ def side_effect(self, url, **kwargs):
...
-        stdout, stderr, _ = run_command(Commands.INSTALL, prefix, "libarchive")
+def test_force_remove(
+    tmp_env: TmpEnvFixture, conda_cli: CondaCLIFixture, monkeypatch: MonkeyPatch
+):
+    with tmp_env() as prefix:
+        monkeypatch.setenv("CONDA_ROOT_PREFIX", prefix)
+        reset_context()
+        stdout, stderr, _ = conda_cli(
+            "install", f"--prefix={prefix}", "libarchive", "--yes"
+        )
```
Comment (kenodegard) [inline tests/test_create.py:2259]: we don't need `CONDA_ROOT_PREFIX` for this test and we can combine the env creation and package installation:

```suggestion
    with tmp_env("libarchive") as prefix:
```

File: tests/test_create.py [hunk 4]
```diff
@@ -2177,94 +2177,105 @@ def side_effect(self, url, **kwargs):
...
         assert package_is_installed(prefix, "libarchive")
         assert package_is_installed(prefix, "xz")
 
-        stdout, stderr, _ = run_command(Commands.REMOVE, prefix, "xz", "--force")
+        stdout, stderr, _ = conda_cli(
+            "remove", f"--prefix={prefix}", "xz", "--force", "--yes"
+        )
```
Comment (kenodegard) [inline tests/test_create.py:2265]:
```suggestion
        conda_cli(
            "remove", f"--prefix={prefix}", "xz", "--force", "--yes"
        )
```

File: tests/test_create.py [hunk 5]
```diff
@@ -2177,94 +2177,105 @@ def side_effect(self, url, **kwargs):
...
         assert not package_is_installed(prefix, "xz")
         assert package_is_installed(prefix, "libarchive")
 
-        stdout, stderr, _ = run_command(Commands.REMOVE, prefix, "libarchive")
+        stdout, stderr, _ = conda_cli(
+            "remove", f"--prefix={prefix}", "libarchive", "--yes"
+        )
```
Comment (kenodegard) [inline tests/test_create.py:2271]:
```suggestion
        conda_cli(
            "remove", f"--prefix={prefix}", "libarchive", "--yes"
        )
```

File: tests/test_create.py [hunk 6]
```diff
@@ -2177,94 +2177,105 @@ def side_effect(self, url, **kwargs):
...
-            stack_callback=conda_tests_ctxt_mgmt_def_pol,
-        ):
-            with pytest.raises(CondaMultiError) as exc:
-                run_command(Commands.INSTALL, prefix, "python")
+def test_disallowed_packages(
+    tmp_env: TmpEnvFixture, conda_cli: CondaCLIFixture, monkeypatch: MonkeyPatch
+):
+    with tmp_env() as prefix:
+        monkeypatch.setenv("CONDA_DISALLOWED_PACKAGES", "sqlite&flask")
+        reset_context()
```
Comment (kenodegard) [inline tests/test_create.py:2185]:
```suggestion
        monkeypatch.setenv("CONDA_DISALLOWED_PACKAGES", "sqlite&flask")
        reset_context()
        assert context.disalloed_packages == ("sqlite", "flask")

```

**Comment by ForgottenProgramme** [comment]
Updated based on reveiw.

**Review by dholth — APPROVED** [review]
LGTM

File: tests/test_create.py [hunk 7]
```diff
@@ -2177,104 +2177,117 @@ def side_effect(self, url, **kwargs):
...
+        assert context.disallowed_packages == ("sqlite", "flask")
+        with pytest.raises(CondaMultiError) as exc:
+            conda_cli("install", f"--prefix={prefix}", "python", "--yes")
         exc_val = exc.value.errors[0]
         assert isinstance(exc_val, DisallowedPackageError)
         assert exc_val.dump_map()["package_ref"]["name"] == "sqlite"
```
Comment (kenodegard) [inline tests/test_create.py:2191]: `conda-forge::python` doesn't depend on `sqlite` so let's use a different package that both `main::python` and `codna-forge::python` depend on:

```suggestion
        monkeypatch.setenv("CONDA_DISALLOWED_PACKAGES", "openssl&flask")
        reset_context()
        assert context.disallowed_packages == ("openssl", "flask")

        with pytest.raises(CondaMultiError) as exc:
            conda_cli("install", f"--prefix={prefix}", "python", "--yes")
        exc_val = exc.value.errors[0]
        assert isinstance(exc_val, DisallowedPackageError)
        assert exc_val.dump_map()["package_ref"]["name"] == "openssl"
```

File: tests/test_create.py [hunk 8]
```diff
@@ -2177,104 +2177,117 @@ def side_effect(self, url, **kwargs):
...
+):
     from conda.core.link import UnlinkLinkTransaction
 
-    with patch.object(UnlinkLinkTransaction, "execute") as mock_method:
-        with make_temp_env("openssl", "--download-only", use_exception_handler=True):
-            assert mock_method.call_count == 0
-        with make_temp_env("openssl", use_exception_handler=True):
-            assert mock_method.call_count == 1
+    with tmp_env() as prefix:
+        spy = mocker.spy(UnlinkLinkTransaction, "execute")
```
Comment (beeankha) [inline tests/test_create.py:2277]: Neat! (`mocker.spy` 👀 )

Comment (kenodegard) [reply tests/test_create.py:2277]: yup! the problem with the standard mock methods is they by default will modify the mocked thing's behavior, `spy` is great if all we need are stats

**Pull request merged**